test/PushNotificationTests/BaseTestSuite.cpp

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include <TestDef.h>
+#include "BaseTestSuite.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Microsoft::Windows::PushNotifications;
+
+void BaseTestSuite::ClassSetup()
+{
+    ::Test::Bootstrap::Setup();
+    bool isSelfContained{ false };
+    if (SUCCEEDED(TestData::TryGetValue(L"SelfContained", isSelfContained)))
```

Comment:
Not a huge fan of this if-check. Why not throw if this fails? I mean we don't want our test to run for non-deterministic behavior right?

Reply:
Unless we are saying that we have a valid scenario here where we don't expect SelfContained to be present as a key?

Reply:
This is basically a switch on the SelfContained scenario. It will run as SelfContained then not SelfContained.

Reply:
But it should never happen where SelfContained isn't set.